Makefile

```diff
@@ -48,18 +47,33 @@ godeps:
 	@echo "skipping godeps"
 endif
 
-build: godeps
-	go build $(PROJECT)/...
+build: godeps go-build
+
+add-patches:
+	cat $(PWD)/patches/*.diff | patch -f -u -p1 -r- -d $(PWD)/../../../
```

Comment:
Maybe create a .PATCHED file marking that the source tree is patched, added (and checked) in add-patches and removed in remove-patches?  This way calling add-patches twice won't break anything

Reply:
Calling add-patches twice will fail to re-apply the patch, but doesn't break anything. It can still be safely removed with remove-patches.